src/bin/cargo/main.rs

```diff
@@ -18,7 +18,7 @@ mod commands;
 use crate::command_prelude::*;
 
 fn main() {
-    setup_logger();
+    let _guard = setup_logger();
```

Comment:
This makes me wonder, should we consider removing and replacing `util::profile` with this? I know @ehuss is using it in `-Zgc` though I never learnt how to use.

There is also an existing [`CARGO_PROFILE`](https://doc.crates.io/contrib/tests/profiling.html#internal-profiler) environment variable documented. We should expand the doc in this PR or a follow-up.

Reply:
@ehuss what are your thoughts on the existing `CARGO_PROFILE` / `util::profile` vs leveraging `tracing` to do profiling?

For myself, having a non-bespoke system to maintain / use and leveraging the same instrumentation for both logging and profiling, seems like a win.

Reply:
I talked with ehuss and he was fine with replacing `CARGO_PROFILE`.  The latest push does this.